iceoryx_utils/include/iceoryx_utils/internal/cxx/string.inl

```diff
@@ -119,9 +119,17 @@ inline string<Capacity>& string<Capacity>::operator=(const char (&rhs)[N]) noexc
     {
```

Comment:
I know you just fixed a bug but wouldn't it be clearer with just a single return and write
```
if (c_str() != rhs)
{
  // add null termination
}
return * this;
```

Reply:
probably, but I wanted to change as less as possible

Reply:
Yeah. Maybe this comment fits better in PR #253 .